library/ssl_tls13_server.c

```diff
@@ -3764,6 +3759,20 @@ static int ssl_write_hello_retry_request( mbedtls_ssl_context *ssl )
     total_ext_len += ext_length + 4  /* 2 bytes for extension_type and 2 bytes for length field */;
 #endif /* MBEDTLS_SSL_COOKIE_C */
 
+    /* Add supported_version extension */
+    if( ( ret = ssl_write_supported_version_ext( ssl,
+                                                 p,
+                                                 ssl->out_buf + MBEDTLS_SSL_OUT_BUFFER_LEN,
+                                                 (size_t *) &ext_length )
```

Comment:
This is a dangerous cast since `ext_length` is an `int` which can have strictly smaller length than `size_t` to which it is casted here. `ext_length` should be a `size_t`, I think.

Reply:
Changed ext_length to size_t